text/text.go

```diff
@@ -4,123 +4,174 @@ import (
 	"bytes"
 	"fmt"
 	"reflect"
+	"time"
 
 	"github.com/ONSdigital/dp-api-clients-go/v2/dataset"
 )
 
-func NewMetadata(m *dataset.Metadata) []byte {
-	var b bytes.Buffer
+func NewMetadata(m *dataset.Metadata, isCustom bool, filterOutputID, downloadServiceURL string) []byte {
+	if isCustom {
+		var b bytes.Buffer
```

Comment:
Could be worth splitting custom dataset metadata and non custom metadata separated into two separate functions? That way easier to test logic